### PR TITLE
none: refine code action regex a bit

### PR DESCRIPTION
**Pull request by Shane-XB-Qian**
it maybe messed digit with str

### CODE REVIEW

**Comment by Shane-XB-Qian** [comment]
it maybe messed digit with str

File: autoload/lsp/codeaction.vim [hunk 1]
```diff
@@ -82,12 +82,13 @@ export def ApplyCodeAction(lspserver: dict<any>, actionlist: list<dict<any>>, qu
 
   var choice: number
 
-  if query =~ '^\d\+'	# digit
-    choice = query->str2nr()
-  elseif query =~ '^/'	# regex
-    choice = 1 + util.Indexof(actions, (i, a) => a.title =~ query[1 : ])
-  elseif query != ''	# literal string
-    choice = 1 + util.Indexof(actions, (i, a) => a.title[0 : query->len() - 1] == query)
+  var query_ = query->trim()
```
Comment (yegappan) [inline autoload/lsp/codeaction.vim:85]: Instead of introducing another "query_" variable, can you reuse query itself here?

**Comment by Shane-XB-Qian** [comment]
> Instead of introducing another "query_" variable, can you reuse query itself here?

I wish, but it cannot.
maybe it's another vim9 script bug/odd thing.

--
shane.xb.qian

**Comment by yegappan** [comment]
> Instead of introducing another "query_" variable, can you reuse query itself here?
> I wish, but it cannot. maybe it's another vim9 script bug/odd thing.

If you use the 'query' variable, do you get an error?  What is the odd thing that you see?

**Comment by Shane-XB-Qian** [comment]
> > > Instead of introducing another "query_" variable, can you reuse query itself here?
> 
> > I wish, but it cannot. maybe it's another vim9 script bug/odd thing.
> 
> If you use the 'query' variable, do you get an error?  What is the odd thing that you see?

yes, "cannot re-use def args", the err.
// have not you tried, or did not know?

-- 
shane.xb.qian

**Pull request merged**